pkg/metrics/percentile/gk_array.go

```diff
@@ -194,6 +194,84 @@ func (s GKArray) interpolatedQuantile(q float64) float64 {
 	return weightBelow*s.Entries[indexBelow].V + weightAbove*s.Entries[indexAbove].V
 }
 
+// Quantiles accepts a slice of quantiles and returns the epsilon estimates of
+// elements at those quantiles.
+func (s GKArray) Quantiles(qValues []float64) []float64 {
+	quantiles := make([]float64, 0, len(qValues))
+	if s.Count == 0 {
+		for _ = range qValues {
+			quantiles = append(quantiles, math.NaN())
+		}
+		return quantiles
+	}
+
+	// This shouldn't happen but checking just in case.
+	if len(s.Incoming) > 0 {
+		s = s.compressWithIncoming(nil)
+	}
+
+	// When there are only a few values just call interpolatedQuantile
+	// for each quantile.
+	if s.Count < int64(1/EPSILON) {
+		for _, q := range qValues {
+			if q < 0 || q > 1 {
+				quantiles = append(quantiles, math.NaN())
+			} else {
+				quantiles = append(quantiles, s.interpolatedQuantile(q))
+			}
+		}
+		return quantiles
+	}
+
+	// By sorting the qValues, the quantiles can be found in one pass
+	// over the Entries
+	sortedQ := make([]float64, 0, len(qValues))
+	// Filter for invalid values
+	for _, q := range qValues {
+		if q >= 0 && q <= 1 {
+			sortedQ = append(sortedQ, q)
+		}
+	}
+	sort.Float64s(sortedQ)
+
+	results := make(map[float64]float64)
+	ranks := make([]int64, 0, len(qValues))
+	for _, q := range sortedQ {
```

Comment:
You could test for sortedness in this loop, or just use `Float64sAreSorted`, or just not check at all